src/action.js

```diff
@@ -0,0 +1,182 @@
+import * as os from "os";
+
+import * as github from "@actions/github";
+import * as toolCache from "@actions/tool-cache";
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+
+// Define consts
+
+export const gruntworkOrg = "gruntwork-io";
+export const patcherRepo = "patcher-cli";
+export const patcherVersion = "v0.4.3";
+
+export const reportCommand = "report";
+export const updateCommand = "update";
+
+export const nonInteractiveFlag = "--non-interactive"
+export const noColorFlag = "--no-color"
+export const skipContainerFlag = "--skip-container-runtime"
+
+function osPlatform() {
+    switch (os.platform()) {
+        case "linux":
+            return "linux";
+        case "darwin":
+            return "darwin";
+        default:
+            core.setFailed("Unsupported operating system - the Patcher action is only released for Darwin and Linux");
+            return;
+    }
+}
+
+export async function openPullRequest(output, dependency, ghToken) {
+    const head = `patcher-updates-${dependency}`
+    const title = `[Patcher] Update ${dependency}`
+    const commitMessage = "Update dependencies using Patcher"
+    const commitAuthor = "Grunty"
+    const commitEmail = "grunty@gruntwork.io"
```

Comment:
Should these be params of the Action? Perhaps with those values as default values?

Reply:
+1 on making these configurable

Reply:
Fixed! There is a default for the https://github.com/gruntwork-patcher user, but can be overwritten with a input.